src/main/scala/mesosphere/marathon/state/FetchUri.scala

```diff
@@ -44,13 +50,18 @@ object FetchUri {
       outputFile = if (uri.hasOutputFile) Some(uri.getOutputFile) else None
     )
 
+  /**
+    * Method check if the given string is a valid URI and if the file type in the
+    * URI is matching a supported file type.
+    *
+    * @param uri String that contain an URI.
+    * @return
+    */
   def isExtract(uri: String): Boolean = {
-    uri.endsWith(".tgz") ||
-      uri.endsWith(".tar.gz") ||
-      uri.endsWith(".tbz2") ||
-      uri.endsWith(".tar.bz2") ||
-      uri.endsWith(".txz") ||
-      uri.endsWith(".tar.xz") ||
-      uri.endsWith(".zip")
+    Try(new URI(uri)).map { u =>
+      supportedFileTypes.exists { fileType =>
+        u.getPath.endsWith(fileType)
```

Comment:
What if the provided uri doesn't have a path which ends as a file but actually serves the right file type?

Reply:
¯\_(ツ)_/¯

Reply:
Just asked @ArmandGrillet how mesos fetcher would handle this.

Reply:
https://github.com/apache/mesos/blob/cc29c27ee597fbe79206d890d695c0a2e64c46aa/src/launcher/fetcher.cpp#L76-L82

It will not be extracted if extension is not matching, but the task will be started, as shown here:

https://github.com/apache/mesos/blob/cc29c27ee597fbe79206d890d695c0a2e64c46aa/src/launcher/fetcher.cpp#L385-L387